include/revng/ABI/CallingConventionTrait.h

```diff
@@ -0,0 +1,1083 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Model/Binary.h"
+#include "revng/Model/Register.h"
+#include "revng/Model/Type.h"
+
+namespace abi {
+
+/// A trait struct used for describing the specifics of the calling convention
+/// for an ABI relevant for revng model.
+///
+/// To register a new ABI for use with the `abi::ABI` template, add
+/// a specialization of this trait.
+template<model::abi::Values V>
+struct CallingConventionTrait {
```

Comment:
Mmmmh, using the non-specialized version of this trait doesn't make much sense and it's probably not what we want.

Maybe we should reframe this as a concept: drop all the members here, leave the non-specialized version undefined (`template<model::abi::Values V> struct CallingConventionTrait;`) and define a `concept` that all the specializations have to respect.

What do you think?

Reply:
I'm all for! Mostly I was using this thing as a "place where all the comments are neatly packed together". The concept should work for that just as well.

Reply:
[Done](https://github.com/revng/revng/pull/98/commits/8268d59c39561de3d5bbc17397dda3983a816121), please let me know what you think!

Reply:
Neat.